drivers/media/i2c/irs1125.c

```diff
@@ -614,8 +613,23 @@ static int irs1125_s_ctrl(struct v4l2_ctrl *ctrl)
 	}
 	case IRS1125_CID_MOD_PLL: {
 		struct irs1125_mod_pll *mod_new;
+		u8 *pval;
+		bool valid;
+		int s;
 
+		s = sizeof(struct irs1125_mod_pll) * IRS1125_NUM_MOD_PLLS;
+		valid = false;
 		mod_new = (struct irs1125_mod_pll *)ctrl->p_new.p;
+
+		for (pval = (u8 *)mod_new; pval < (u8 *)mod_new + s; pval++) {
+			if (*pval != 0) {
+				valid = true;
+				break;
+			}
+		}
```

Comment:
I'm a little confused by the validity test.
If any one 8bit value within an array of 8 16bit values is non-zero, then the whole thing is valid?

So all of these are all equally valid
0x0000 0x0000 0x0000 0x0001 0x0000  0x0000  0x0000 0x0000 
0xdead 0xbeef 0xdead 0xbeef 0xdead 0xbeef 0xdead 0xbeef
0x0000 0x0000 0x0000 0x000 0x0000  0x0000  0x0000 0x8000 
0x8000 0x0000 0x0000 0x000 0x0000  0x0000  0x0000 0x0000 
I'll believe you. Do you really need to constrain this because the device dies if all 0's are set, or can you just let userspace worry about setting something sane?

And why test as 8bit values? The same comparison can be done on the u16 values, again treated as an array.

Reply:
I have a conceptual problem here. I would also like userspace to worry about setting something sane, but IRS1125_CID_MOD_PLL is a compound control. I want to keep controls and hardware in sync which I guess is a key concept of V4L2 (I'll remove all unnecessary V4L2_CTRL_FLAG_EXECUTE_ON_WRITE).

Now the problem is, that when userspace calls a "close" on the video device, the imager goes into deep sleep, which I guess is a good idea to safe power. But in this state, it also looses all its register contents and the V4L2 controls need to be synced with the HW again after the wakeup.
I do that inside __sensor_init with v4l2_ctrl_handler_setup(...). Here the IRS1125_CID_MOD_PLL contains a default value of "all zeros" which kills the imager. I'll look deeper into that problem, I think that the pre-devider value of 0 might be the problem, which would make the code much clearer here.

Anyway I did not find any other solution to that problem. Maybe you have one...? I think a p_def member to v4l2_ctrl_config is added to more recent kernel versions (https://lore.kernel.org/patchwork/patch/1147819/) which would give me the possibility to add a senseful default value for IRS1125_CID_MOD_PLL.

Reply:
OK, so you need IRS1125_CID_MOD_PLL initialised to something, or skip it if it hasn't been set.

Do you also need to guard against userspace setting all 0's? This will show up as a standard V4L2 device, so if all zeros is really fatal (hardware damage fatal?) then the driver ought to guard against it.
Minor nitpick - ```if (*pval) {``` is neater than ```if (*pval != 0) {```
It would also be nice if the sensor would just stream with default values, but I realise that this may not be terribly useful.

If you rebase against 5.4, then I'm happy for those patches from 5.5 to be cherry-picked back (probably ignore the vivid ones). We do insert a line 
``` Commit <hash> upstream```
into the commit text when cherry-picking back, just to make maintenance easier.
You can then set a default :-)

Reply:
You can look at the current change. I think it is a good idea, that the userspace is responsible for writing senseful data in this control. I skip writing this control only in case of the internal sync after HW power up.